lib/actions.js

```diff
@@ -0,0 +1,361 @@
+     * Then it installs sources and
+     * dependencies in the cozy-light folder.
+     *
+     * @param {String} plugin Plugin to install
+     *                        (ex: cozy-labs/cozy-light-docker).
+     * @param {Function} callback Termination.
+     */
+    installPlugin: function (plugin, callback) {
+      LOGGER.info('Fetching data...');
+      npmHelpers.fetchManifest(plugin,
+        function addPluginToConfig (fetchErr, manifest, type) {
+          if (fetchErr) {
+            LOGGER.raw(fetchErr);
+            LOGGER.error('Cannot find given plugin manifest.');
+            LOGGER.error('Make sure it lives on Github');
+            LOGGER.error('or in the given directory.');
+            LOGGER.error(plugin + ' installation failed.');
+            nodeHelpers.invoke(callback, fetchErr);
+          } else {
+            var config = configHelpers.getConfig();
+            var pluginName = manifest.name;
+            if ( config.plugins[pluginName] ) {
+              LOGGER.info('Plugin ' + pluginName + ' already installed...');
+              return nodeHelpers.invoke(callback, true, manifest, type);
+            } else {
+              LOGGER.info('Installing plugin ' + pluginName + '...');
+              var setupCb = function(installErr){
+                if (installErr) {
+                  LOGGER.raw(installErr);
+                  LOGGER.error(pluginName + ' installation failed.');
+                  nodeHelpers.invoke(callback, installErr, manifest, type);
+                } else {
+                  LOGGER.info(pluginName + ' installed. Enjoy!');
+                  configHelpers.watcher.one(function(){
+                    pluginHelpers.loadPlugin(pluginName);
+                    nodeHelpers.invoke(callback);
+                  });
+                  configHelpers.addPlugin(pluginName, manifest);
+                }
+              };
+              if (type === 'file') {
+                return npmHelpers.link(plugin, setupCb);
+              } else if (type === 'url') {
+                return npmHelpers.install(plugin, setupCb);
+              }
+            }
+          }
+        });
+    },
+
+    /**
+     * Remove plugin from config and its source from node module folder.
+     *
+     * @param {String} plugin Plugin to remove.
+     * @param {Function} callback Termination.
+     */
+    uninstallPlugin: function (plugin, callback){
+      var config = configHelpers.getConfig();
+      LOGGER.info('Removing ' + plugin + '...');
+      if (config.plugins[plugin] === undefined) {
+        LOGGER.error(plugin + ' is not installed.');
+        nodeHelpers.invoke(callback, true);
+      } else {
+        var module = config.plugins[plugin].name;
+        npmHelpers.uninstall(module, function removePluginFromConfig (err) {
+          if (err) {
+            LOGGER.raw(err);
+            LOGGER.error('npm did not uninstall ' + plugin + ' correctly.');
+            nodeHelpers.invoke(callback, err);
```

Comment:
Did you mean

```

cozyLight.configHelpers = require('./config-helper')();

cozyLight.nodeHelpers = require('./node-helpers');
cozyLight.npmHelpers = require('./npm-helpers')(cozyLight);
cozyLight.pluginHelpers = require('./plugin-helpers')(cozyLight);
cozyLight.applicationHelpers = require('./application-helpers')(cozyLight);
cozyLight.mainAppHelper = require('./main-app-helper')(cozyLight);
cozyLight.actions = require('./actions')(cozyLight);
```

?

Need to do that to declare first the structure of cozyLight object. Then i can pass it to the sub modules.
If you don t do that, you are sending a partial object instead (something related to internal javascript/v8 functionning), and that don t comply the expectations.